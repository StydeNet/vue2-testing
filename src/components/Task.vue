<template>
  <div
    class="list-group-item d-flex justify-content-between align-items-center"
    :class="{ disabled: task.done }"
  >
    <div>
      <router-link :to="{ path: '/task', params: { task } }"></router-link>
      {{ task.name }}<input class="ml-3 extra-class" v-model="isChecked" type="checkbox">
    </div>
    <span class="badge badge-pill badge-danger" id="delete" @click="$emit('delete')">
      <slot name="close"></slot>
    </span>
    <slot></slot>
  </div>
</template>
<script>
import { mapActions } from 'vuex';

export default {
  name: 'Task',

  props: {
    task: Object
  },

  computed: {
    isChecked: {
      get() {
        if (!this.task) return false;
        return this.task.done;
      },

      set(value) {
        if (value) {
          this.completeTask(this.task);
        } else {
          this.uncompleteTask(this.task);
        }
      }
    }
  },

  methods: {
    ...mapActions(['completeTask', 'uncompleteTask'])
  }
};
</script>
