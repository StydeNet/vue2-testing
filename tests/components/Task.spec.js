import { mount, shallow } from '@vue/test-utils';
import Task from '@/components/Task';

describe('Component Task', () => {
  test('it has name', () => {
    const wrapper = mount(Task);
    expect(wrapper.name()).toBe('Task');
  });

  test('it renders task prop', () => {
    const wrapper = mount(Task);
    wrapper.setProps({ task: 'My new Task' });
    expect(wrapper.text()).toContain('My new Task');
  });

  test('it emmits delete event when delete button is clicked', () => {
    const wrapper = mount(Task);
    const button = wrapper.find('#delete');
    button.trigger('click');
    expect(wrapper.emitted().delete).toBeTruthy();
  });

  test('it renders default slot', () => {
    const wrapper = shallow(Task, {
      slots: {
        default: 'DEFAULT SLOT'
      }
    });
    expect(wrapper.text()).toContain('DEFAULT SLOT');
  });

  test('it renders close slot', () => {
    const wrapper = shallow(Task, {
      slots: {
        close: 'CLOSE SLOT'
      }
    });
    expect(wrapper.find('#delete').text()).toContain('CLOSE SLOT');
  });
});
