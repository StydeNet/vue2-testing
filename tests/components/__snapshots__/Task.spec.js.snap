// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Task renders correctly 1`] = `
"<div class=\\"list-group-item d-flex justify-content-between align-items-center\\"><div><div to=\\"[object Object]\\"></div>
    My Task<input type=\\"checkbox\\" class=\\"ml-3 extra-class\\"></div> <span id=\\"delete\\" class=\\"badge badge-pill badge-danger\\"></span> </div>"
`;
